internal/sveltos/status.go

```diff
@@ -25,13 +25,14 @@ import (
 	hmc "github.com/Mirantis/hmc/api/v1alpha1"
 )
 
-// SetStatusConditions transforms status from Sveltos ClusterSummary
-// object and sets it into the provided list of conditions.
-func SetStatusConditions(summary *sveltosv1beta1.ClusterSummary, conditions *[]metav1.Condition) error {
+// SetStatusConditions creates & sets a list of conditions from provided ClusterSummary and returns it.
+func SetStatusConditions(summary *sveltosv1beta1.ClusterSummary) ([]metav1.Condition, error) {
 	if summary == nil {
-		return errors.New("nil summary provided")
+		return nil, errors.New("error getting status from ClusterSummary: nil summary provided")
 	}
 
+	conditions := []metav1.Condition{}
```

Comment:
nits: i think we can preallocate the `conditions` slice with something like `conditions := make(..., 0, len(summary.Status.FeatureSummaries)` and probably change the function's name from `Set` to `Get` or similar (because now it does not modify the given slice of conditions)

Reply:
Preallocated `conditions`. For the name, we decided to keep it `Set`, please see convo above.

Reply:
Ah, now I see, yes, it looks like it’s better to keep Get as Michael suggested and as you did initially. I was a little bit confused by the `SetStatusCondition`. @wahabmk  Let’s bring the initial version if you don’t mind.

Reply:
@zerospiel I'll merge this PR as is to close the task but update the function name in a separate PR.